### PR TITLE
fix: avoid panic on invalid inputs for ecdsa verification

### PR DESCRIPTION
**Pull request by guipublic**
# Description

## Problem

Resolves https://github.com/noir-lang/noir/issues/11058 and #11059 

## User Documentation

Check one:
- [X] No user documentation needed.
- [ ] Changes in _docs/_ included in this PR.
- [ ] **[For Experimental Features]** Changes in _docs/_ to be submitted in a separate PR.

# PR Checklist

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: acvm-repo/blackbox_solver/src/ecdsa/secp256k1.rs [hunk 1]
```diff
@@ -96,10 +94,27 @@ pub(super) fn verify_signature(
         + (ProjectivePoint::from(*pubkey.as_affine()) * u2))
         .to_affine();
 
+    // Compare R.x with signature's r component.
     match R.to_encoded_point(false).coordinates() {
-        Coordinates::Uncompressed { x, y: _ } => Ok(Scalar::from_repr(*x).unwrap().eq(&r)),
+        Coordinates::Uncompressed { x, y: _ } => {
+            // The conversion from R.x to a scalar can fail if R.x >= curve_order (a possible but rare case).
+            // In this case, the signature is invalid per ECDSA specification, so we return false.
+            // The prover will handle this gracefully - it should generate a proof that fails verification.
```
Comment (TomAFrench) [inline acvm-repo/blackbox_solver/src/ecdsa/secp256k1.rs:102]: Have we confirmed that barretenberg follows this? A test in aztec-packages dsl folder would be nice.

Comment (guipublic) [reply acvm-repo/blackbox_solver/src/ecdsa/secp256k1.rs:102]: It seems barrentenberg should even generate a valid proof (that the signature verification is false), because it does not crash on invalid R.x:
```
    // Step 8.
    // We reduce result.x() to 2^s, where s is the smallest s.t. 2^s > q. It is cheap in terms of constraints, and
    // avoids possible edge cases
```

Creating a test for this case does not look easy.

**Comment by TomAFrench** [comment]
@guipublic there seems to be a couple failing tests as well, not sure if you're aware.

**Comment by guipublic** [comment]
> @guipublic there seems to be a couple failing tests as well, not sure if you're aware.

I see, we don't panic anymore and some tests were expecting a panic. That's good, some more parameters to test bb with!

**Comment by guipublic** [comment]
I updated and moved the failing test cases to execution_success, I also added some inputs for the invalid R.x case
I checked they can be proven and verified successfully with BB (in the secpk1 case).
Having them in execution_success will have BB's CI to pick and run them.
